src/main/java/duke/command/AddItemCommand.java

```diff
@@ -0,0 +1,91 @@
+package duke.command;
+
+import duke.exception.DukeException;
+import duke.storage.Storage;
+import duke.task.Deadline;
+import duke.task.DukeTask;
+import duke.task.Event;
+import duke.task.ToDo;
+import duke.tasklist.TaskList;
+import duke.ui.Ui;
+
+/**
+ * Encapsulates adding items to a TaskList object.
+ *
+ * @author Kartikeya
+ */
+public class AddItemCommand implements Command {
+    private final String[] input;
+
+    /**
+     * Constructor for an AddItemCommand object.
+     *
+     * @param input split strings input by user
+     */
+    public AddItemCommand(String[] input) {
+        this.input = input;
+    }
+
+    /**
+     * {@inheritDoc}
+     * Adds given input item to itemList, and shows the resulting message to the user.
+     */
+    @Override
+    public void execute(TaskList itemList, Ui ui, Storage storage) throws DukeException {
```

Comment:
For the naming of method, good job on using a verb! But perhaps you might want to name it as "executeSomething"

Reply:
The problem with that is the same `execute` method is overriden by every `...Command` class, which makes it hard to find a consistent name. Perhaps executeCommand? But that seems redundant.